src/collections/collection.ts

```diff
@@ -33,8 +31,38 @@ import { logger } from '@/src/logger';
 // https://github.com/mongodb/node-mongodb-native/pull/3323
 type JSONAPIUpdateResult = Omit<UpdateResult, 'upsertedId'> & { upsertedId: ObjectId | null };
 
-interface DocumentCallback {
-  (err: Error | undefined, res: any): void;
+export interface FindOneOptions {
+  sort?: Record<string, 1 | -1>;
```

Comment:
As per current spec of JSON API, findOne API doesn't take `options` input. Please see https://github.com/stargate/jsonapi/blob/main/docs/jsonapi-spec.md#findone-command-options

Reply:
But, I think it's ok to leave it on the driver side, since the JSON API might throw an error back now when some try to pass the options, but will ll be good when the JSON API eventually starts accepting `options`.

Reply:
I originally called these `JSONAPIFindOneOptions`, etc. but decided against it because, like you said, the driver layer's options are slightly different from the jsonapi's options. So you can think of these interfaces as the user-facing definitions for what options the driver layer expects, and then the driver layer figures out how to convert those options into the structure that jsonapi expects.

Reply:
Makes sense.